ci/sync-protobuf.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+set -eoux pipefail
+
+# syn-protobuf.sh is a bash script to sync the protobuf
+# files using the proto-compiler project. It will check
+# out the protobuf files from the git versions specified in
+# proto/src/prost/COSMOS_SDK_COMMIT and
+# proto/src/prost/COSMOS_IBC_COMMIT. If you want to sync
+# the protobuf files to a newer version, modify the
+# relevant files with the new commit IDs.
+
+# This script should be run from the root directory of ibc-rs
+
+# We can specify where to clone the git repositories
+# for cosmos-sdk and ibc-go. By default they are cloned
+# to /tmp/cosmos-sdk.git and /tmp/ibc-go.git.
+# We can override this to existing directories
+# that already have a clone of the repositories,
+# so that there is no need to clone the entire
+# repositories over and over again every time
+# the script is called.
+
+COSMOS_SDK_GIT=${COSMOS_SDK_GIT:-/tmp/cosmos-sdk.git}
+IBC_GO_GIT=${IBC_GO_GIT:-/tmp/ibc-go.git}
```

Comment:
That could work as well. Compared to the original, we want to use a local git bare repository as cache instead of trying to clone the entire remote repository every time the command is run.

Right now I still have the problem that if a bare repository has already been cloned before, I can't find simple way to update it other than having to reclone the entire bare repository. So it is still a little problematic to set a persistent location at `$HOME` by default.

Reply:
Use of non-randomized paths in `/tmp` also assumes that only a single user will ever run this script (over the host's uptime before `/tmp` is cleared). This is probably OK for a typical developer's workstation, but using a location at `$HOME` by convention gives you better isolation.

> Right now I still have the problem that if a bare repository has already been cloned before, I can't find simple way to update it

I've tried locally and `git fetch` or `git fetch origin` seem to work. What's the problem?

Reply:
> I've tried locally and `git fetch` or `git fetch origin` seem to work. What's the problem?

Hmm I might have cloned the bare repository inappropriately. That seems to work now!

I have updated the script as you suggested. It stores the git cache at `~/.cache/cosmos`.

Reply:
> It stores the git cache at `~/.cache/cosmos`.

This might conceivably clash with some Cosmos client in the future (or just an unrelated app named `cosmos`), I'd rather use something identifying the ibc-rs build scripts as the well-known directory under `~/.cache` and put everything cached by the scripts in there. But this is mostly pedantry.